seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/psql/PostgresJdbcRowConverter.java

```
@@ -316,4 +362,110 @@ public String microsecondsToIntervalFormatVal(String intervalVal) {
         if (seconds > 0) sb.append(seconds).append(" seconds");
         return sb.toString().trim();
     }
+
+    private OffsetDateTime getPostgresOffsetDateTime(ResultSet rs, int columnIndex)
+            throws SQLException {
+        // Read the value ONCE to avoid drivers returning null on subsequent reads
+        final Object obj;
+        try {
+            obj = rs.getObject(columnIndex);
+        } catch (SQLException e) {
+            log.debug("Failed to get object from ResultSet at column {}", columnIndex, e);
+            // Best-effort fallback to string, still only a single read path for parsing
+            try {
+                final String str = rs.getString(columnIndex);
+                if (str != null && !str.trim().isEmpty()) {
+                    return parsePostgresTimestampTz(str);
+                }
+            } catch (SQLException se) {
+                log.debug("Failed to get string from ResultSet at column {}", columnIndex, se);
+            }
+            return null;
```

`getPostgresOffsetDateTime`/`parsePostgresTimestampTz` 's fault-tolerant strategy is "parse failed to return null", and only debug-level logging is done. This means that once a timestamptz string is legal but not overwritten by the current regex (e.g. with time-distinguished name, lowercase z, or other equivalent formats returned by future drivers), SeaTunnel will silently write the field as null. 
This should throw SQLException when it cannot be parsed, so that the task fails quickly with the "data unparsed" error, rather than silently swallowing it

---

All timestamptz parsing goes through a single normalized path—ISO parse first, then UTC fallback. Only when both fail do we throw an SQLException with the original string, so unsupported formats no longer get silently written as null.